### PR TITLE
Fix table row indices to handle headers and footers correctly

### LINKED ISSUE #66: Table row indices not starting from 0 when using headers.
When using the `ShadTable` component with headers, the row indices do not start from 0. This causes only a subset of rows to be displayed correctly. The issue occurs because the effective row count and row indices are not correctly adjusted for the presence of headers and footers.

**Steps to Reproduce**:
1. Use the `ShadTable` component with a header.
2. Observe that the first row starts from 1 instead of 0.

**Expected Behavior**:
The row indices should start from 0 even when a header is present.

### PR DESCRIPTION
**Pull request by Kyziq**
This pull request addresses the issue where row indices do not start from 0 when using headers in the `ShadTable` component. The changes ensure that the effective row count is calculated correctly and that row indices are adjusted accordingly.

**Changes**:
- Adjusted `effectiveRowCount` to account for headers and footers.
- Modified `cellBuilder` to correctly adjust row indices when a header is present.

**Issue Reference**:
closes #66 

**Testing**:
Tested with a table that includes headers and footers to ensure all rows are displayed correctly.

### CODE REVIEW

File: lib/src/components/table.dart [hunk 1]
```diff
@@ -214,8 +214,8 @@ class _ShadTableState extends State<ShadTable> {
 
   int get effectiveRowCount =>
       widget.rowCount +
-      (widget.header != null ? 1 : 0) +
-      (widget.footer != null ? 1 : 0);
+      (widget.headerBuilder != null ? 1 : 0) +
+      (widget.footerBuilder != null ? 1 : 0);
```
Comment (nank1ro) [inline lib/src/components/table.dart]: Thanks for the PR @Kyziq 🙏
Here the code was partially correct, you must check both `widget.header` and `widget.headerBuilder`

**Review by nank1ro — CHANGES_REQUESTED** [review]

File: lib/src/components/table.dart (hunk 1 above)
Comment (nank1ro) [reply lib/src/components/table.dart]: This is needed to have the same behavior when using `ShadTable.list` constructor

Comment (kyziq) [reply lib/src/components/table.dart]: Thanks for pointing that out! I've updated the code to check both `widget.header` and `widget.headerBuilder` for consistent behavior across both constructors, including `ShadTable.list`. This ensures correct handling of headers and footers

**Review by nank1ro — APPROVED** [review]
LGTM

**Pull request merged**